src/core/vectortile/qgsvectortilebasiclabeling.cpp

```diff
@@ -170,7 +174,11 @@ bool QgsVectorTileBasicLabelProvider::prepare( QgsRenderContext &context, QSet<Q
     QgsExpressionContextScopePopper popper( context.expressionContext(), scope );
 
     mSubProviders[i]->setFields( fields );
-    mSubProviders[i]->prepare( context, attributeNames );
+    if ( !mSubProviders[i]->prepare( context, attributeNames ) )
+    {
+      QgsDebugMsg( QStringLiteral( "Failed to prepare labeling for style index" ) + QString::number( i ) );
```

Comment:
Is that something to send to the message log?

Reply:
I don't think so - prepare() fails e.g. if the labeling expression is invalid, which is something that happens normally while editing labeling expression...

Reply:
Isn't editing labeling expressions something that's done in released versions where debug messages are not visible?

Reply:
Correct. This is just a message for developers, meaning that something may have gone wrong... Or maybe I am missing your point :-)

Reply:
Let's see if I can be more precise :)
How will "I" as someone configuring a project realize that my expression has an error did not work? I guess often this will be triggered by faulty expressions (and not by source code problems a developer needs to take care of)?

Reply:
Your labels won't show up - and the expression will be red in the GUI... just like with labeling of vector layers...

Reply:
Yeath, true, that's the same amount of information as in this debug message :+1: